sql/datetimetype.go

```diff
@@ -60,6 +60,13 @@ var (
 	//
 	// https://github.com/MariaDB/server/blob/mysql-5.5.36/sql-common/my_time.c#L124
 	TimestampDatetimeLayouts = []string{
+		"2006-01-02 15",
```

Comment:
I'm a little on the fence about some of MySQL's wackier time formats. I could go either way on a few of these. "YYYY-MM-DD HH" and "YYYY-MM-DD HH:" are the two that seem the wackiest. All the others seem fine to me though.  

My vote is hold back on these first two unless customers specifically ask for them, but I could be convinced to keep them in.